### PR TITLE
Add XCTS system and tests

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Add the XCTS system and verifies the Schwarzschild solution solves it numerically.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
I could move all commits except the first to a separate PR, but I kept them here for now because they test the XCTS system is implemented correctly.

**Comment by nilsvu** [comment]
@haraldp271 and @nilsdeppe Would you be willing to continue the XCTS system review in this PR?

**Comment by nilsdeppe** [comment]
sure, but I'm going to wait until the dependent PRs are merged since it's unclear which commits I should be reviewing

File: src/Elliptic/Systems/Xcts/FluxesAndSources.cpp
Comment (nilsdeppe) [inline src/Elliptic/Systems/Xcts/FluxesAndSources.cpp]: [optional] in these empty functions, maybe add a brief comment here why they're empty? Just so that when someone looks at the source file there's a quick reminder

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (nilsdeppe) [inline src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: missing `\` in front of `Gamma`

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (nilsdeppe) [inline src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: considering breaking this up over two lines

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (nilsdeppe) [inline src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: considering breaking this up over two lines

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (nilsdeppe) [inline src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: 4 missing `\` in front of `gamma`s

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (nilsdeppe) [inline src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: question, is there a "Right" value of `n` or does that depend on either the physical setup or the matter type?

**Review by nilsdeppe — COMMENTED** [review]
LGTM, much easier to review than expected! Some dox+comment suggestions. Please feel free to squash immediately :)

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (nilsvu) [reply src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: 👍 reformatted these equations

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (nilsvu) [reply src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: I added a note to the docs. https://arxiv.org/abs/0804.3787 suggests n=6 to make volume integrals independent of the conformal factor (see footnote 1). @haraldp271 would you like to add some insight? :)

**Comment by nilsvu** [comment]
Rebased and squashed in the changes to the docs. Also added the small commit "Fix a typo in longitudinal operator docs". Thanks for reviewing @nilsdeppe !

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
Rebased and added a fixup commit that verifies the freshly merged Kerr solution (#3068) :)

**Review by nilsdeppe — DISMISSED** [review]
LGTM

**Comment by nilsvu** [comment]
Rebased again. @haraldp271 Could you review this when you find the time?

File: src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp
Comment (haraldp271) [reply src/Elliptic/Systems/Xcts/FirstOrderSystem.hpp]: The big thing is to have n big enough for the correct sign.  Beyond that, I am not aware of a strongly preferred value (n=6 is nice, as the  integrated matter density independent of conformal factor)

**Review by haraldp271 — APPROVED** [review]
Looks good to me.  I keep on being surprised by the large amount of scaffolding codes that the templates necessitate.

**Pull request merged**
